bridge_adaptivity/module/models.py

```diff
@@ -110,6 +113,86 @@ def get_absolute_url(self):
         return reverse('module:collection-list')
 
 
+class Engine(models.Model):
+    """Defines engine settings."""
+
+    DEFAULT_ENGINE_NAME = 'mock'
+
+    name = models.CharField(default=DEFAULT_ENGINE_NAME, max_length=255)
+    host = models.URLField(blank=True, null=True)
+    token = models.CharField(max_length=255, blank=True, null=True)
+
+    _engines_cache = {}
+
+    class Meta:
+        unique_together = (('host', 'token'),)
+
+    def __str__(self):
+        return "Engine: {}".format(self.name)
+
+    @classmethod
+    def get_default_engine(cls):
+        if not cls.objects.count():
```

Comment:
I would use `get_or_create` or you see any restriction to use this shortcut method? BTW I'm not sure that `mock` engine should be default one I think we need to add the flag `default` which could be True only for one engine.

Reply:
I added this fag in other branch